source/Calamari.Tests/KubernetesFixtures/Authentication/SetupKubectlAuthenticationAwsFixture.cs

```diff
@@ -0,0 +1,336 @@
+
+        private IVariables variables;
+        private ILog log;
+        private ICommandLineRunner commandLineRunner;
+        private IKubectl kubectl;
+        private ICalamariFileSystem fileSystem;
+        private Dictionary<string, string> environmentVars;
+
+        private SetupKubectlAuthenticationFixture.Invocations invocations;
+
+        // TODO: Strip down/commonize
+        [SetUp]
+        public void Setup()
+        {
+            invocations = new SetupKubectlAuthenticationFixture.Invocations();
+            invocations.AddLogMessageFor("which", "kubelogin", "kubelogin");
+            invocations.AddLogMessageFor("where", "kubelogin", "kubelogin");
+            AddLogForAwsVersion(CurrentAwsVersion);
+
+            variables = new CalamariVariables();
+
+            log = Substitute.For<ILog>();
+            commandLineRunner = Substitute.For<ICommandLineRunner>();
+            commandLineRunner.Execute(Arg.Any<CommandLineInvocation>())
+                .Returns(
+                    x =>
+                    {
+                        var invocation = x.Arg<CommandLineInvocation>();
+                        var isSuccess = true;
+                        string logMessage = null;
+                        if (invocation.Executable != "chmod")
+                        {
+                            isSuccess = invocations.TryAdd(invocation.Executable, invocation.Arguments, out logMessage);
+                        }
+                        if (logMessage != null)
+                            invocation.AdditionalInvocationOutputSink?.WriteInfo(logMessage);
+                        return new CommandResult(
+                            invocation.Executable,
+                            isSuccess ? 0 : 1,
+                            workingDirectory: workingDirectory);
+                    });
+            kubectl = Substitute.For<IKubectl>();
+            kubectl.ExecutableLocation.Returns("kubectl");
+            kubectl.TrySetKubectl().Returns(true);
+            kubectl.When(x => x.ExecuteCommandAndAssertSuccess(Arg.Any<string[]>()))
+                .Do(
+                    x =>
+                    {
+                        var args = x.Arg<string[]>();
+                        if (args != null)
+                            invocations.TryAdd("kubectl", string.Join(" ", args), out var _);
+                    });
+            fileSystem = Substitute.For<ICalamariFileSystem>();
+
+            variables.Set(SpecialVariables.ClusterUrl, AwsClusterUrl);
+            variables.Set(SpecialVariables.EksClusterName, EksClusterName);
+            variables.Set(SpecialVariables.Namespace, Namespace);
+
+            environmentVars = new Dictionary<string, string>
+            {
+                { "AWS_ACCESS_KEY_ID", "access_key" },
+                { "AWS_SECRET_ACCESS_KEY", "secret_key" },
+                { "AWS_REGION", "region" }
+            };
+        }
```

Comment:
I'll commonise some of this with the main `SetupKubectlAuthentication` tests once I've re-applied my changes over Rob's refactoring